src/signals/incident-management/containers/IncidentSplitContainer/IncidentSplitContainer.js

```diff
@@ -30,11 +30,28 @@ const IncidentSplitContainer = ({ FormComponent }) => {
   const dispatch = useDispatch();
   const [parentIncident, setParentIncident] = useState();
   const [directingDepartment, setDirectingDepartment] = useState([]);
+  const [note, setNote] = useState();
   const departments = useSelector(makeSelectDepartments);
   const directingDepartments = useSelector(makeSelectDirectingDepartments);
 
   const [subcategoryGroups, subcategoryOptions] = useSelector(makeSelectSubcategoriesGroupedByCategories);
 
+  const parentIncidentPatchData = useMemo(() => {
+    if (!parentIncident) return {};
+
+    const shouldPatchDirectingDepartment =
```

Comment:
Long name for a boolean. This construction is difficult to understand. Can we think about something more readable? Maybe destructure the `parentIncident`?

Reply:
Aside from my comments on the lines below, I find this okay to read.
You cold make it a bit more readable by using some more consts along the way, like this:

```
const directingDepartments = parentIncident.directing_departments;
const differentLength = directingDepartments.length !== directingDepartment.length;
const differentValues = differentLength || !directingDepartments.some(department => department?.id === directingDepartment[0]?.id);
const shouldPatch = !directingDepartments || differentValues;
```